packages/core/src/interpreter.ts

```diff
@@ -502,6 +503,15 @@ export class Interpreter<
       ? this.behavior.getSnapshot(this._state)
       : this._state;
   }
+
+  public reset(
+    persistedState: PersistedStateFrom<TBehavior> | undefined = this.options
+      .state
+  ): void {
+    this.stop();
```

Comment:
Can we skip this or do a soft stop that doesn't unsubscribe existing listeners?

Reply:
I'm going to change this so that resetting is a separate procedure that doesn't unsubscribe.

Reply:
Sounds good. This is going to be really helpful.